src/lib/libast/tm/tmxfmt.c

```diff
@@ -137,7 +137,6 @@ tmxfmt(char* buf, size_t len, const char* format, Time_t t)
 	{
 		if ((c = *format++) == delimiter)
 		{
-			delimiter = 0;
```

Comment:
I'm not comfortable with this deletion. Are you sure it's correct?

Reply:
It should be harmless. The `for(;;)` loop it's in is rather gigantic, so it can be hard to follow. The logic behind is pretty simple though. It starts [here](https://github.com/ksh93/ksh/blob/24d5cd853e3661c704abb55ed596d03d51c4e722/src/lib/libast/tm/tmxfmt.c#L136-L138) at L136-138, and when the break at [L140-142](https://github.com/ksh93/ksh/blob/24d5cd853e3661c704abb55ed596d03d51c4e722/src/lib/libast/tm/tmxfmt.c#L140-L142) happens it skips to the code after the for loop, at [L711-715](https://github.com/ksh93/ksh/blob/24d5cd853e3661c704abb55ed596d03d51c4e722/src/lib/libast/tm/tmxfmt.c#L711-L715). If this happens, then `delimiter` is not used after the `break;`. If that condition isn't fulfilled, then the code at [L143-146](https://github.com/ksh93/ksh/blob/24d5cd853e3661c704abb55ed596d03d51c4e722/src/lib/libast/tm/tmxfmt.c#L143-L146) sets `delimiter` to the value of `sp->delimiter`, making the initial `delimiter = 0` redundant.

Reply:
Thanks. Okay, LGTM.